From 7ca2ede8e4338749be0a560776ea96e9387f43ec Mon Sep 17 00:00:00 2001
From: Aniket Limaye <a-limaye@ti.com>
Date: Sun, 22 Jan 2023 20:45:14 +0530
Subject: [PATCH 4/4] arm64: dts: ti: k3-j721e-rtos-memory-map: CMA DT node

- Adding in a DT node to reserve 512MB for CMA allocation in the upper space
- Carrying until kernel command line based solution is feasible:
  https://jira.itg.ti.com/browse/LCPD-32468

Signed-off-by: Aniket Limaye <a-limaye@ti.com>
---
 arch/arm64/boot/dts/ti/k3-j721e-rtos-memory-map.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/ti/k3-j721e-rtos-memory-map.dtsi b/arch/arm64/boot/dts/ti/k3-j721e-rtos-memory-map.dtsi
index 76c1a65da5..e8324b84ea 100644
--- a/arch/arm64/boot/dts/ti/k3-j721e-rtos-memory-map.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-j721e-rtos-memory-map.dtsi
@@ -235,6 +235,12 @@
 		reg = <0x08 0x80000000 0x00 0x27000000>;
 		no-map;
 	};
+	linux_cma_region: linux-cma-buffers@8a7000000 {
+		compatible = "shared-dma-pool";
+		reusable;
+		reg = <0x08 0xa7000000 0x00 0x20000000>;
+		linux,cma-default;
+	};
 };
 
 &mcu_r5fss0_core0 {
-- 
2.17.1

